docs/source/index.rst

```diff
@@ -10,7 +10,26 @@
 JupyterLab Documentation
 ########################
 
-JupyterLab is the next-generation web-based user interface for Project Jupyter. `Try it on Binder <https://mybinder.org/v2/gh/jupyterlab/jupyterlab-demo/3818244?urlpath=lab/tree/demo>`__.  JupyterLab follows the Jupyter `Community Guides <https://jupyter.readthedocs.io/en/latest/community/content-community.html>`__.
+Welcome to the **JupyterLab** documentation site. **JupyterLab** is a highly
+extensible, feature-rich notebook authoring application and editing environment,
+and is a part of `Project Jupyter <https://docs.jupyter.org/en/latest/>`_, a
+large umbrella project centered around the goal of providing tools for interactive
+computing with `computational notebooks <https://docs.jupyter.org/en/latest/#what-is-a-notebook>`_.
+
+A `computational notebook <https://docs.jupyter.org/en/latest/#what-is-a-notebook>`_
+is a shareable document that combines computer code, plain language descriptions,
+data, rich visualizations like 3D models, charts, graphs and figures, and interactive
+controls. A notebook, along with an editor like **JupyterLab**, provides a fast interactive
```

Comment:
Would "development environment" characterize JupyterLab more precisely than editor? References:
1. https://jupyter.org/
> JupyterLab is the latest web-based interactive development environment for notebooks, code, and data.
 2. https://github.com/jupyterlab/jupyterlab#readme
 > An extensible environment for interactive and reproducible computing, based on the Jupyter Notebook and Architecture.

Reply:
As discussed in this thread https://github.com/jupyter/notebook/pull/6877#discussion_r1201285583, if the goal here is to reduce word count and simplify the description of Notebook, "editor" is a good option. Alternative options I propose above don't communicate their relation to notebook as clearly.